flow360/component/utils.py

```diff
@@ -62,22 +62,66 @@ def match_file_pattern(patterns, filename):
     return False
 
 
+def _valid_resource_id(resource_id):
+    """
+    returns:
+    1. Whether the resource_id is valid
+    2. the prefix of the resource_id
+    3. the uuid part of the resource_id
+    """
+    if isinstance(resource_id, str) is False:
+        raise ValueError(f"resource_id must be a string, but got {type(resource_id)}")
+
+    pattern = re.compile(
+        r"""
+        ^                      # Start of the string
+        (?:(?P<prefix>         # Start of the optional prefix group
+        [0-9a-zA-Z\-]{1,16}    # Prefix: 1 to 16 characters, alphanumeric or dash
+        )-)?
+        (?P<uuid>              # Start of the UUID group
+        [0-9a-fA-F]{8}-        # 8 hex digits, followed by a dash
+        [0-9a-fA-F]{4}-        # 4 hex digits, followed by a dash
+        [0-9a-fA-F]{4}-        # 4 hex digits, followed by a dash
+        [0-9a-fA-F]{4}-        # 4 hex digits, followed by a dash
+        [0-9a-fA-F]{12}        # 12 hex digits
+        )$
```

Comment:
I think there was a miscommunication, the 16 length was not for prefix itself but for for entire id. I think we can be less strict with this validation since it doesn't matter much what is the format of ID

Reply:
so basically it is no longer a UUID?

Reply:
Done